README.md

```diff
@@ -196,6 +196,57 @@ specify "when event is not found" do
 end
 ```
 
+To test line used (email, push, pigeon), you can the following shared examples:
+
+```ruby
+##
+# Line checker for `active_delivery`
+#
+# ==== Attributes
+#
+# * +lines+ - Array of possible lines (:mailer, :pusher, :pigeon..)
+# * +methods+ - methods called on each line
+shared_examples_for "deliverable" do |lines, methods|
+  it "calls correct lines and methods" do
+    lines.each do |line|
+      methods.each do |method|
+        expect(described_class.delivery_lines[line]).to receive(:notify).with(method, anything).once
+        subject
+      end
+    end
+  end
+end
+
+##
+# Line non-checker for `active_delivery`
+#
+# ==== Attributes
+#
+# * +lines+ - Array of possible lines (:mailer, :pusher)
+# * +methods+ - methods called on each line
+shared_examples_for "not_deliverable" do |lines, methods|
+  it "calls correct lines and methods" do
+    lines.each do |line|
+      methods.each do |method|
+        expect(described_class.delivery_lines[line]).not_to receive(:notify).with(method, anything)
+        subject
+      end
+    end
+  end
+end
+```
+
+and call them like that:
+
+```ruby
+RSpec.describe V1::MyDelivery do
+  context "when..." do
+    it_behaves_like "deliverable", [:mailer], [:some_action]
```

Comment:
I like the idea.

Can we make the API a bit more explicit? For example:

```ruby
# single line
it_behaves_like "deliverable", via: :mailer, action: :some_action
# multiple lines
it_behaves_like "deliverable", via: [:mailer, :pusher], action: :some_action
```

Btw, what is the case for multiple actions? I think, the most common case is to check for a single action; hence, for multiple actions we can just include the shared example multiple times.

Reply:
Sorry I forgot this one! I'll take a look asap. 🙂 

For the multiple actions, it was only to avoid to copy/paste the same shared example.

Reply:
No worries. I think, we can actually merge this.